src/org/zaproxy/zap/extension/stats/ExtensionStats.java

```diff
@@ -0,0 +1,163 @@
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.stats;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.extension.OptionsChangedListener;
+import org.parosproxy.paros.model.OptionsParam;
+import org.zaproxy.zap.extension.api.API;
+import org.zaproxy.zap.utils.Stats;
+
+public class ExtensionStats extends ExtensionAdaptor implements OptionsChangedListener {
+
+	public static final String NAME = "ExtensionStats"; 
+	
+	private InMemoryStats inMemStats = null;
+	private Statsd statsd = null;
+	private OptionsStatsPanel optionsStatsPanel = null;
+	private StatsAPI statsAPI = new StatsAPI();
+	private StatsParam statsParam = null;
+
+    private Logger logger = Logger.getLogger(ExtensionStats.class);
+
+	public ExtensionStats() {
+		super();
+		initialize();
+	}
+
+	private void initialize() {
+        this.setName(NAME);
+	}
+
+	@Override
+	public void hook(ExtensionHook extensionHook) {
+	    super.hook(extensionHook);
+
+		extensionHook.addOptionsChangedListener(this);
+		extensionHook.addOptionsParamSet(getStatsParam());
+
+	    if (getView() != null) {
+	        extensionHook.getHookView().addOptionPanel(getOptionsStatsPanel());
+	    }
+	    
+	    API.getInstance().registerApiImplementor(statsAPI);
+	    statsAPI.addApiOptions(getStatsParam());
+
+	}
+
+	@Override
+	public void optionsLoaded() {
+		optionsChanged();
+	}
+
+	private OptionsStatsPanel getOptionsStatsPanel() {
+		if (optionsStatsPanel == null) {
+			optionsStatsPanel = new OptionsStatsPanel(this);
+		}
+		return optionsStatsPanel;
+	}
+	
+	private StatsParam getStatsParam() {
+		if (statsParam == null) {
+			statsParam = new StatsParam();
+		}
+		return statsParam;
+	}
+	
+	@Override
+	public boolean isCore() {
+		return true;
+	}
+	
+	@Override
+	public String getAuthor() {
+		return Constant.ZAP_TEAM;
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString("stats.desc");
+	}
+
+	@Override
+	public URL getURL() {
+		try {
+			return new URL(Constant.ZAP_HOMEPAGE);
+		} catch (MalformedURLException e) {
+			return null;
+		}
+	}
+
+	/**
+	 * No database tables used, so all supported
+	 */
+	@Override
+	public boolean supportsDb(String type) {
+    	return true;
+    }
+
+	@Override
+	public void optionsChanged(OptionsParam optionsParam) {
+		optionsChanged();
+	}
+	
+	private void optionsChanged() {
+        if ((inMemStats == null) == this.getStatsParam().isInMemoryEnabled()) {
```

Comment:
The check `inMemStats == null` could be extracted to a variable (it could be reused in the following if statement and the code would be easier to read).